omnibus/config/templates/datadog-agent/systemd.sysprobe.service.erb

```diff
@@ -1,6 +1,7 @@
 [Unit]
 Description=Datadog System Probe
 Requires=sys-kernel-debug.mount
+Before=datadog-agent.service
```

Comment:
You'll want to include

```
Wants=datadog-agent.service
```

as well.  From https://www.freedesktop.org/software/systemd/man/systemd.unit.html

> Before=, After=
> ...
> Note that those settings are independent of and orthogonal to the requirement dependencies as configured by Requires=, Wants=, Requisite=, or BindsTo=. It is a common pattern to include a unit name in both the After= and Wants= options, in which case the unit listed will be started before the unit that is configured with these options.

Reply:
Wouldn't `BindTo` be better than `Wants`? If I'm not mistaken, that would make the system-probe restart when datadog-agent restarts.

https://www.freedesktop.org/software/systemd/man/systemd.unit.html#BindsTo=

Reply:
@leeavital My reasoning was to keep the two services as loosely coupled as possible. This way starting system-probe doesn't necessarily starts the agent (not sure if there would be an use case for this, though).

@albertvaka Would it be a desirable behaviour to restart system-probe every time datadog-agent is restarted? E.g. if you change a setting on one of the agent integrations, in theory we wouldn't need to restart system-probe?

Reply:
The system-probe has no use if the agent isn't running, so I think it makes sense to tie it to the agent. 

About restarting them together: IMO, the fact that they are separate processes is an implementation detail, so as a user it makes sense to treat everything in the Agent as a single entity and restart everything together. We already do this with the process and trace agents, which are also separate binaries.